doc/Manuals/ug-ss_x-road_6_security_server_user_guide.md

```diff
@@ -2143,7 +2144,39 @@ For example, these log messages are related to an API call with correlation ID `
 
 ### 19.4 Validation errors
 
-An error response from the REST API can include validation errors if an unsupported parameter was provided with the request. 
+An error response from the REST API can include validation errors if an unsupported parameter was provided with the request.
+When 
+
+Example request and response of adding a new subsystem with illegal characters:
+```
+POST https://ss1:4100/api/v1/clients
+
+Request body:
+{
+  "client": {
+    "member_class": "ORG",
+    "member_code": "0/1234",
+    "subsystem_code": "Subsystem%Code"
+  },
+  "ignore_warnings": false
+}
+
+Response body:
+{
+  "error": {
+    "code": "validation_failure",
+    "validation_errors": {
+      "clientAdd.client.memberCode": [
+        "NoForwardslashes"
+      ],
+      "clientAdd.client.subsystemCode": [
+        "NoPercents"
+      ]
+    }
+  },
+  "status": 400
+}
+```
```

Comment:
The following lines are not part of this PR, but are related to acceptance criteria, and anyway since we're already at the topic....

> In addition to the validation messages declared in Java Validation API, the following validation errors are possible

How about a table with simple one sentence explanations (maybe just the defaultMessage) about each custom validation error type? Granted most are pretty self-explanatory, but `Normalized` is probably not.

Also, is the list up to date? `IdentifierValidationErrorInfo` seems to think there is `NoControlChars` instead of `Normalized`?

Reply:
It would also be good to make "Java Validation API" a proper external reference

Reply:
What do you think about adding that table, or at least what "NoControlChars" specifically means?

```
    /**
     * checks if the string contains ISO control characters or zero-width spaces
     */
    public static boolean containsControlChars(String s) {
        return CharMatcher.javaIsoControl().matchesAnyOf(s) || s.indexOf(FORBIDDEN_BOM) >= 0
                || s.indexOf(FORBIDDEN_ZWSP) >= 0;
    }
```

Reply:
Good idea. Added a table where the errors are explained. Ended up explaining the obvious ones too but it's better now